discovery-provider/alembic/versions/6b5186e7d28f_replace_aggregate_plays_mat_view_with_.py

```diff
@@ -0,0 +1,541 @@
+"""Replace aggregate_plays mat view with table
+
+Revision ID: 6b5186e7d28f
+Revises: 36eac5ed00bf
+Create Date: 2021-11-19 20:54:17.596441
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '6b5186e7d28f'
+down_revision = '36eac5ed00bf'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    conn = op.get_bind()
+    query = """
+        DO
+        $do$
+        BEGIN
+            -- only run migration if the aggregate_plays table does not already exist
+            -- trying to drop the mat view aggregate_plays will fail if the table aggregate_plays exists
+            -- also don't want to risk losing data
+            IF NOT EXISTS (
+                SELECT FROM information_schema.tables
+                WHERE table_name   = 'aggregate_plays'
+            ) THEN
+
+                -- create aggregate_plays table
+                DROP TABLE IF EXISTS aggregate_plays_table;
+                CREATE TABLE aggregate_plays_table AS TABLE aggregate_plays;
+
+                -- drop existing aggregate_plays mat view and its dependencies
+
+                DROP MATERIALIZED VIEW IF EXISTS trending_params;
+                DROP MATERIALIZED VIEW IF EXISTS aggregate_plays; -- will complain if aggregate_plays table exists
```

Comment:
what does this comment mean?

Reply:
going off the comments above, these steps are only run if the **table**  aggregate_plays doesn't exist. otherwise, postgres will be confused and think we're trying to drop the table not a mat view.